### PR TITLE
Updates azure-sdk-bom Version in Preparation for Release

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: sdk/boms/azure-sdk-bom/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.azure</groupId>
   <artifactId>azure-sdk-bom</artifactId>
-  <version>1.0.0-beta.1</version>
+  <version>1.0.0</version>
```
Comment (srnagar) [inline sdk/boms/azure-sdk-bom/pom.xml:8]: Should this have a version tag? Makes it easy for e2e and spring modules to refer to the bom and update when this gets updated.

Comment (alzimmermsft) [reply sdk/boms/azure-sdk-bom/pom.xml:8]: I can add one, the BOM isn't version managed in the same way the other modules are, @JimSuplizio thoughts?

**Comment by alzimmermsft** [comment]
> Should we consider adding our direct dependencies to the BOM (JacksonXML, etc), at least temporarily?

https://github.com/Azure/azure-sdk-for-java/issues/10342

File: sdk/boms/azure-sdk-bom/pom.xml (hunk 1 above)
Comment (JimSuplizio) [reply sdk/boms/azure-sdk-bom/pom.xml:8]: The answer is no, there will be no tags in the bom file. First, the bom file was only supposed to be released periodically (every 3-6 months according to the original conversation) and second, my automation specifically disallows dependencyManagement sections which, you know, is pretty much what a bom file is.

**Pull request merged**